### ISSUE TITLE
RStudio Desktop causes a crash in the shiny app, complaining about closure coersion to a character vector

### ISSUE THREAD

**Issue by ashokkrish** [ashokkrish/episim#87]
```
Warning: Error in as.character: cannot coerce type 'closure' to vector of type 'character'
  1: shiny::runApp
```

Could one of you please take a look?

**Comment by kle6951** [comment]
Hi Ashok, I ran into the same issue before with this current version of the code. I talked with Bryce to inspect the issue and it turned out to be the problem with RStudio. I am able to run it normally when you open the application with VSCode. It also works normally with other text editors. Right now, we are not sure why it happens only with RStudio.

**Comment by ashokkrish** [comment]
@kle6951 Thanks for your note. Can one of you post it on StackOverflow or something? My conference workshop is coming up soon and I don't want to switch to a new IDE this late. I am using R version 4.3.2 and the RStudio version is

<img width="195" alt="image" src="https://github.com/ashokkrish/episim/assets/410949/96f16479-e720-4aaa-a363-fbc296de0ac9">

**Comment by bryce-carson** [comment]
Similarly to #15, and the discussion we had in our document on Google Docs related to app.R/global.R, the issue has to do with what RStudio looks for when it is asked to run an app.

There is no actual issue with the application, with episim, so I'm labelling this as "won't fix," becasue I _can't_ fix it. There might be something we can do, but it would affect any other _entirely legitimate and valid way of launching a Shiny app_. Emacs, VS Code, Shiny Server (I think) each simply run an R script.

In Emacs I source the global.R file. In VS Code, @Toby-exe sources the same file, or issues the command `runApp()` while the working directory is the root of the application. @Toby-exe, so you're aware, the proper way to launch the application would be to call `shinyAppDir(here())`, as I wrote in the bottom of global.R; however, I too also run the command `runApp()` sometimes instead and I haven't noticed a difference.

@ashokkrish, the solution will lie in studying the function documentation for `shiny::runApp()`, `shiny::shinyAppDir()`, and also reading RStudio Desktop documentation to see what algorithm RStudio is following which is causing the breakage.

In summary, it is the fault of RStudio Desktop. For now I'll leave this issue open, even though I could close it and we can keep commenting in this issue, but it'll be easier to find for you, @ashokkrish, if I leave it open, so I will. There's not really anything to fix right now, until someone determines what RStudio Desktop is bungling.

**Comment by bryce-carson** [comment]
> @kle6951 Thanks for your note. Can one of you post it on StackOverflow or something? My conference workshop is coming up soon and I don't want to switch to a new IDE this late. I am using R version 4.3.2 and the RStudio version is
> <img alt="image" width="195" src="https://private-user-images.githubusercontent.com/410949/334623883-96f16479-e720-4aaa-a363-fbc296de0ac9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY5NDg3NDAsIm5iZiI6MTcxNjk0ODQ0MCwicGF0aCI6Ii80MTA5NDkvMzM0NjIzODgzLTk2ZjE2NDc5LWU3MjAtNGFhYS1hMzYzLWZiYzI5NmRlMGFjOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUyOVQwMjA3MjBaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMWJlYWEzZjdhMjNhNTVlYTMzZWQyNjRhOTk2NmUyNWU3YTAxNDFhODQ0ZmJhZTllODBhMDJmYzM2NDcyZWY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.7A8l5L0g2IRfCmds9JnL-S_d7_c1Kjjcb-OPJQrkPC8">

A quick fix which I just discovered, which works in RStudio, @ashokkrish, is to change the way the app is launched. Note that I am able to reproduce the issue you're experiencing in RStudio Desktop on Linux.

Pictured is the change you must make, which is to run the application in a background job, not in the RStudio Console (the source of the issue). This works on Linux and it _must_ work on other platforms as well. A "background job" in RStudio Desktop's terminology is closer to what Emacs and VS Code use, however they are communicating with an interactive R process. RStudio Desktop normally does the same, but the way its communication with the interactive R process takes place is different and is bungling the application.

![Screenshot at 2024-05-28 20-09-40](https://github.com/ashokkrish/episim/assets/51974245/0d5a4ee3-0593-44ec-827e-ce177af8cb64)

Because this is a working fix, and is a "two click" fix, I'm going to close the issue. Please reopen the issue if it doesn't work on Windows, @ashokkrish.

**Comment by ashokkrish** [comment]
@bryce-carson That did the trick. I am now able to run the code without errors.

**Comment by bryce-carson** [comment]
> @bryce-carson That did the trick. I am now able to run the code without errors.

Glad to hear. I'll unpin the issue.

**Comment by bryce-carson** [comment]
@ashokkrish, I don't know what changed, but one of my previous commits "resolved" this issue. You can now press "Run App" with the default "In R Console" selected in the dropdown menu again, and the application will work.

This is current as of https://github.com/ashokkrish/episim/commit/52b8a1f60ae5135d2aa14675d661abff2ca9ce0a.
